### DISCUSSION TITLE
zstd memory cost

### DISCUSSION THREAD

**Discussion by Nyx2022** [klauspost/compress#655] [Q&A]
hello,everyone,  when i use zstad.SpeedBestCompression with multi times to compresss multi []byte objectes that are different sizes,i found the memory cost will increase to a large number, so i wanna know is  there a method to get the best compression and keep a low memory cost,thakn you !

**Comment by klauspost** [comment]
You can use [WithEncoderConcurrency(1)](https://pkg.go.dev/github.com/klauspost/compress/zstd#WithEncoderConcurrency) to only allocate a single encoder.

This means only a single EncodeAll call will be able to run concurrently, but otherwise it shouldn't have any downsides.

**Reply by Nyx2022** [reply]
> You can use [WithEncoderConcurrency(1)](https://pkg.go.dev/github.com/klauspost/compress/zstd#WithEncoderConcurrency) to only allocate a single encoder.
> 
> This means only a single EncodeAll call will be able to run concurrently, but otherwise it shouldn't have any downsides.

yes，it's a way! actually,i wanna both,effency and low cost

**Reply by klauspost** [reply]
@Nyx2022 Wouldn't we all?
